### PR TITLE
feat(crux): standard health check

### PR DESCRIPTION
**Pull request by robot9706**
- Implemented the standard gRPC health check based on https://github.com/grpc/grpc/blob/master/doc/health-checking.md
- Used `grpc_health_probe` for testing (https://github.com/grpc-ecosystem/grpc-health-probe)

### CODE REVIEW

File: protobuf/proto/health.proto [hunk 1]
```diff
@@ -0,0 +1,24 @@
+syntax = "proto3";
```
Comment (polaroi8d) [inline protobuf/proto/health.proto:1]: We want to create it independently and not add to the existing services?

Comment (robot9706) [reply protobuf/proto/health.proto:1]: The standard says that the health check service should be in the 'grpc.health.v1' package, so I think this is the only way I can do that, no?

File: web/crux/src/app/health/health.controller.ts [hunk 2]
```diff
@@ -4,16 +4,65 @@ import { CruxHealthController, CruxHealthControllerMethods, HealthResponse } fro
...
+  constructor(private readonly service: HealthService, private mapper: HealthMapper) {}
 
   @AuditLogLevel('disabled')
   async getHealth(): Promise<HealthResponse> {
     return this.service.getHealth()
   }
+
+  @AuditLogLevel('disabled')
+  async check(request: HealthCheckRequest): Promise<HealthCheckResponse> {
+    if (!request.service || request.service === '') {
```
Comment (polaroi8d) [inline web/crux/src/app/health/health.controller.ts:33]: Why do you need to check the `request.service` property falseness?

File: web/crux/src/app/health/health.controller.ts [hunk 3]
```diff
@@ -4,16 +4,65 @@ import { CruxHealthController, CruxHealthControllerMethods, HealthResponse } fro
...
+      const health = await this.service.getHealth()
+      return {
+        status: this.mapper.serviceStatusToProto(health.status),
+      }
+    }
+    if (request.service === 'db') {
```
Comment (polaroi8d) [inline web/crux/src/app/health/health.controller.ts:39]: I know this was created earlier but if the `crux` service is down we can't get accurate information from the Database based on the dependencies. At present, it's okay but in the long run, we have to rework this.

File: web/crux/src/app/health/health.controller.ts [hunk 4]
```diff
@@ -4,16 +4,65 @@ import { CruxHealthController, CruxHealthControllerMethods, HealthResponse } fro
...
+      const health = await this.service.getHealth()
+      return {
+        status: health.lastMigration
+          ? HealthCheckResponse_ServingStatus.SERVING
+          : HealthCheckResponse_ServingStatus.NOT_SERVING,
+      }
+    }
+    if (request.service === 'kratos') {
+      const health = await this.service.getKratosHealth()
```
Comment (polaroi8d) [inline web/crux/src/app/health/health.controller.ts:48]: You could extract the common variables like `health` outside of the if-else blocks. Minor add new lines between ifs IMHO.

**Review by polaroi8d — CHANGES_REQUESTED** [review]

File: web/crux/src/app/health/health.controller.ts (hunk 4 above)
Comment (robot9706) [reply web/crux/src/app/health/health.controller.ts]: Rearranged the code a bit.

File: web/crux/src/app/health/health.controller.ts [hunk 5]
```diff
@@ -4,16 +4,62 @@ import { CruxHealthController, CruxHealthControllerMethods, HealthResponse } fro
...
+  constructor(private readonly service: HealthService, private mapper: HealthMapper) {}
 
   @AuditLogLevel('disabled')
   async getHealth(): Promise<HealthResponse> {
     return this.service.getHealth()
   }
+
+  @AuditLogLevel('disabled')
+  async check(request: HealthCheckRequest): Promise<HealthCheckResponse> {
+    if (request.service === '' || request.service === 'db') {
```
Comment (m8vago) [inline web/crux/src/app/health/health.controller.ts:33]: this logic should be inside the service itself

File: web/crux/src/app/health/health.controller.ts (hunk 2 above)
Comment (robot9706) [reply web/crux/src/app/health/health.controller.ts]: I used a standard tool to check the implementation and it sends nothing to check the service health (the standard says empty string so I implemented both).

File: web/crux/src/app/health/health.controller.ts [hunk 6]
```diff
@@ -4,16 +4,50 @@ import { CruxHealthController, CruxHealthControllerMethods, HealthResponse } fro
...
+  @AuditLogLevel('disabled')
+  async check(request: HealthCheckRequest): Promise<HealthCheckResponse> {
+    const status = await this.service.getServiceStatus(request.service)
+    if (status) {
+      return {
+        status: this.mapper.serviceStatusToProto(status),
+      }
+    }
+
+    throw new NotFoundException({
```
Comment (m8vago) [inline web/crux/src/app/health/health.controller.ts:39]: move it inside to the serivice

**Review by m8vago — CHANGES_REQUESTED** [review]

**Review by m8vago — APPROVED** [review]

**Pull request merged**